pkg/aggregator/spandecorator/decorator.go

```diff
@@ -12,49 +12,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package eventdecorator
+package spandecorator
 
 import (
 	"context"
 
-	"github.com/kubewharf/kelemetry/pkg/aggregator/aggregatorevent"
+	"github.com/kubewharf/kelemetry/pkg/aggregator/tracer"
 	"github.com/kubewharf/kelemetry/pkg/manager"
 	"github.com/kubewharf/kelemetry/pkg/util"
 )
 
 func init() {
-	manager.Global.Provide("event-union-decorator", manager.Ptr[UnionEventDecorator](&unionDecorator{}))
+	manager.Global.Provide("event-union-decorator", manager.Ptr[UnionSpanDecorator](&unionDecorator{}))
 }
 
 type Decorator interface {
-	Decorate(ctx context.Context, object util.ObjectRef, event *aggregatorevent.Event)
+	Decorate(ctx context.Context, object util.ObjectRef, span *tracer.Span)
```

Comment:
Is it really a good idea to expose the low-level `tracer` object to the higher-level decorators? Resource tags feel like  high-level logic that shouldn't touch the implementation about generating the trace.

Reply:
it is the only way to add tag for a object span, if we don't add tag for object span, it must make sure that every audit and event has beed tagged, it will take too many calls to kube-apiserver

Reply:
maybe we can pass through tags as decorate input parameter, then tracer.Span won't be exposed

Reply:
What about adding a separate interface for object spans?